testing/integration/testdata/.upgrade-test-agent-versions.yml

```diff
@@ -5,6 +5,8 @@
 # upgrade integration tests.
 
 testVersions:
+  - 9.0.2-SNAPSHOT
```

Comment:
this one shouldn't be here, we have a bug 🙂

Reply:
Hey @pkoutsovasilis, just trying to understand: why should this version not be here?

Reply:
So AFAICT, the testVersion are filtered and sorted based on these criteria

```
{
  "UpgradeToVersion": "9.0.2",
  "CurrentMajors": 1,
  "PreviousMajors": 1,
  "PreviousMinors": 2,
  "SnapshotBranches": [
    "9.0",
    "8.19",
    "7.17"
  ]
}
```

9.0.2-SNAPSHOT and 9.0.1 are not previous minors and for that reason TestPreviousMinor is failing [here](https://buildkite.com/elastic/elastic-agent/builds/21324#0196ed5a-a51c-4538-aacb-a49a4eba79fe/144-609).